include/neml2/tensors/Vec.h

```diff
@@ -25,29 +25,31 @@
 #pragma once
 
 #include "neml2/tensors/VecBase.h"
+#include "neml2/models/crystallography/SymmetryTransformable.h"
 
 namespace neml2
 {
 class Scalar;
 class R2;
 class Rot;
+namespace crystallography
+{
+class SymmetryOperator;
+}
 
 /**
  * @brief The (logical) vector.
  *
  * The logical storage space is (3).
  */
-class Vec : public VecBase<Vec>
+class Vec : public VecBase<Vec>, public crystallography::SymmetryTransformable<Vec>
```

Comment:
Can we avoid the diamond pattern using
```
class Vec : public crystallography::SymmetryTransformable<VecBase<Vec>>
```
?

The same comment applies to all classes deriving from `SymmetryTransformable`.

Reply:
Sure.